README.md

```diff
@@ -2,92 +2,133 @@
+3. Tunable Resource Requirements
+  - Individual nodes should be able to control the amount of machine resources (HDD/CPU/RAM) they must expend to be part of the network.
+
+These design principles are aimed at ensuring that participation in the Portal Network is feasible even on resource constrained devices.
+
+## The JSON-RPC API
 
 The following JSON-RPC API endpoints are intended to be supported by the portal network and exposed by portal clients.
 
 TODO
 
+## Bridge Nodes
+
+We use the term "bridge" node to refer to the process by which data enters the Portal Network.  A "bridge" node is a normal node in the Portal Network that is able to bring in new and missing data from some external source and inject it into the Portal Network.  These nodes would not be special in any way from the perspective of the Portal Network.
+
+The planned architecture for bridge nodes is to pull data from the standard JSON-RPC API of a Full Node and "push" this data into their respective networks within the Portal Network.
+
 ## Network Functionality
 
-### State: Accounts and Contract Storage
+### State Network: Accounts and Contract Storage
+
+The State Network facilitates on-demand retrieval of the Ethereum "state" data.  This includes things like:
 
-The state network facilitates on-demand retrieval of the Ethereum "state" data.
+- Reading account balances or nonce values
+- Retrieving contract code
+- Reading contract storage values
 
-Nodes should be able to choose how much state they want to store and share, and the network should provide a way to identify which nodes to query for a wanted portion of state. This is so that every node, no matter how small, can contribute to the health and robustness of the network.
+The responsibility for storing the underlying "state" data should be evenly distributed across the nodes in the network.  Nodes must be able to choose how much state they want to store.  The data is distributed in a manner that allows nodes to determine the appropriate nodes to query for any individual piece of state data.  When retrieving state data, a node should be able to validate the response using a recent header from the header chain.
 
-The network will be dependent on receiving new and updated state for new blocks. Full "bridge" nodes acting as benevolent state providers would be responsible for bringing in this data from the main network. The network should be able to remain healthy even with a small number of bridge nodes.
+The network will be dependent on receiving new and updated state for new blocks. Full "bridge" nodes acting as benevolent state providers would be responsible for bringing in this data from the main network. The network should be able to remain healthy even with a small number of bridge nodes.  As new data enters the network, nodes must be able to validate the data using a recent header from the header chain.
```

Comment:
In the intro paragraph future tense was converted to present tense. Do you want to make the same change here? "would be responsible" -> "are responsible"

Reply:
"As new data enters the network, nodes must be able to validate the data using a recent header from the header chain." what does this "must" mean? I'm used to specs using ["must"](https://www.ietf.org/rfc/rfc2119.txt) to mean something required in order for an implementation of the spec to count as a correct implementation.

Does this mean that in order to participate in the state network nodes must have access to the header chain, either by being a full node or by participating in the history network?

Reply:
updated from "must be able" to "are able" because yes, a node could just choose to not verify the data and the network would not care.